docs/guide/writing-tasks.md

```diff
@@ -154,6 +154,10 @@ test_task:
     Tasks for PRs upload caches to a separate caching namespace to not interfere with caches used by other tasks.
     But such PR tasks **can read** all caches even from the main caching namespace for a repository.
 
+!!! warning "Scope of cached artifacts"
+    Cache artifacts are shared between tasks, so two caches with the same name on e.g. Linux and OSX containers will share the same set of files.
+    This may introduce binary incompatibility between caches. To avoid that, add `uname -ms` into `fingerprint_script` which will distinguish caches based on OS and machine architecture.
```

Comment:
Use `CIRRUS_OS `:

```suggestion
    This may introduce binary incompatibility between caches. To avoid that, add `echo $CIRRUS_OS` into `fingerprint_script` which will distinguish caches based on OS.
```

Reply:
Thanks! TIL about suggestions on GitHub.

Do you think `CIRRUS_OS` alone would be sufficient? For binary compatibility, machine type (like x86, amd64, ARM etc) may also be important.

Reply:
At the moment I think it's enough. Cirrus supports amd64 only at the moment. 🤔

Reply:
I don’t really see a point in having 32 bit anyway.

Reply:
Test some IOT stuff 🤷‍♂️ There is some interest in #218

Reply:
Ah.

Reply:
Done!